### PR TITLE
Updated various versions of 3rd-party libraries and tools

### PR DESCRIPTION
**Pull request by sormuras**
Replaced deprecated API calls with new ones.

### CODE REVIEW

File: src/test/java/com/squareup/javapoet/CodeBlockTest.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
       CodeBlock.builder().add("$1>", "taco").build();
       fail();
     } catch (IllegalArgumentException exp) {
-      assertThat(exp).hasMessage("$$, $>, $<, $[, $], and $W may not have an index");
+      assertThat(exp).hasMessageThat().isEqualTo("$$, $>, $<, $[, $], and $W may not have an index");
```
[the remark this answers is not in the document]
Comment (vanniktech) [reply src/test/java/com/squareup/javapoet/CodeBlockTest.java]: Yup hasMessage is deprecated in truth

File: src/test/java/com/squareup/javapoet/FileReadingTest.java [hunk 2]
```diff
@@ -113,7 +113,8 @@
     assertThat(diagnosticCollector.getDiagnostics()).isEmpty();
 
     ClassLoader loader = fileManager.getClassLoader(StandardLocation.CLASS_OUTPUT);
-    Callable<?> test = Class.forName("foo.Test", true, loader).asSubclass(Callable.class).newInstance();
```
[the remark this answers is not in the document]
Comment (sormuras) [reply src/test/java/com/squareup/javapoet/FileReadingTest.java]: On my way.

[the remark this answers is not in the document]
Comment (sormuras) [reply src/test/java/com/squareup/javapoet/FileReadingTest.java]: Why is checkstyle not configured to check the test source folder for bad style? When I activated it on the "JUnit 5" PR it revealed a lot of warnings/errors.

File: src/test/java/com/squareup/javapoet/CodeBlockTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (swankjesse) [reply src/test/java/com/squareup/javapoet/CodeBlockTest.java]: Ugh. Do these API authors even use their APIs? Ick.

**Pull request merged**